boards/BMS/CMakeLists.txt

```diff
@@ -37,17 +37,52 @@ list(REMOVE_ITEM GOOGLETEST_TEST_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/Test/Src/main
 set(GOOGLETEST_TEST_INCLUDE_DIRS
         "${CMAKE_CURRENT_SOURCE_DIR}/Test/Inc")
 
+# The App and Io source files must be sorted in two platform-specific lists:
+#
+# 1. ARM_BINARY_X86_COMPATIBLE_SRCS - These can be compiled on x86
+# 2. ARM_BINARY_X86_INCOMPATIBLE_SRCS - These can't be compiled on x86
+#
+# Every App file can be compiled on x86. However, the same is not true for Io
+# files. Some Io files rely on STM headers (e.g. stm32f3xx.h) and thus can't be
+# compiled on x86. The rest of the Io files, however, can be compiled on x86.
+#
+# These relationships can be summarized as follows:
+#
+# ARM_BINARY_X86_COMPATIBLE_SRCS = App source files + x86-compatible Io files
+# ARM_BINARY_X86_INCOMPATIBLE_SRCS = Io sources files - x86-compatible Io files
+#
+# On the other hand, the App and Io include directories are always
+# x86-compatible because they are paths and not actual files. We just have to
+# place App and Io include directories in ARM_BINARY_INCLUDE_DIRS.
```

Comment:
This really feels like we should have it in `boards/CMakeLists.txt`, rather then duplicated in the per-board `CMakeLists.txt`.

Reply:
Do you mean just the comments?

Reply:
I'm not super set on leaving the comments on a per-board basis though. Just wanted to let you know what my thinking process was first.

Reply:
Yeah, I think in this case the duplication is significant enough that moving it to `boards/CMakeLists.txt` is worth it, even if it might not be totally correct in terms of keeping things isolated.

Reply:
Added it as a `@note` under `create_arm_binary_or_tests_for_board`